Python/specialize.c

```diff
@@ -517,6 +533,13 @@ analyze_descriptor(PyTypeObject *type, PyObject *name, PyObject **descr, int sto
         if (desc_cls->tp_flags & Py_TPFLAGS_METHOD_DESCRIPTOR) {
             return METHOD;
         }
+        if (Py_IS_TYPE(descr, &PyClassMethodDescr_Type)) {
```

Comment:
This should be `descriptor` not `descr`

Reply:
What the heck ... that fixed the stack overflow on ASAN.

Reply:
Oh right, Py_TYPE reads into ob_type which is out of bounds. Thanks a bunch! You just saved me from an hour of headaches.

Reply:
`descr` is a `PyObject **`, not a `PyObject *`, it points to a local variable in the caller.

Reply:
I feel somewhat dumb to have missed that now. Thank you. Though I'm more confused as to why MSVC didn't complain. Accessing random fields on something that isn't a struct usually kills compilation.

Oh, I see it now: `_Py_IS_TYPE` is a static inline function, not a macro, so the compiler wouldn't have known.